cmake/leveldb.cmake

```diff
@@ -74,21 +74,23 @@ target_include_directories(leveldb
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/leveldb/include>
 )
 
+add_library(nowarn_leveldb_interface INTERFACE)
 if(MSVC)
-  target_compile_options(leveldb
-    PRIVATE
-      /wd4244
-      /wd4267
-      $<$<CONFIG:Release>:/wd4722>
+  target_compile_options(nowarn_leveldb_interface INTERFACE
```

Comment:
Why are these no longer private?

Reply:
Only `INTERFACE_*` properties might be populated for [interface libraries](https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries).

The `nowarn_leveldb_interface` library itself is linked in the `PRIVATE` scope though.

Reply:
Otherwise, CMake raises an error:
```
CMake Error at cmake/leveldb.cmake:86 (target_compile_options):
  target_compile_options may only set INTERFACE properties on INTERFACE
  targets
Call Stack (most recent call first):
  CMakeLists.txt:260 (include)

```